internal/metrics/datanode.go

```diff
@@ -0,0 +1,240 @@
+
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+const (
+	// TODO: use the common status label
+	DataNodeMetricLabelSuccess = "success"
+	DataNodeMetricLabelFail    = "fail"
+	DataNodeMetricLabelTotal   = "total"
+
+	DataNodeMsgTypeInsert = "insert"
+	DataNodeMsgTypeDelete = "delete"
+)
+
+// TODO: move to metrics.go
+const nodeIDLabelName = "node_id"
+
+// TODO: move to metrics.go
+func GetCollectionLabel(collectionID typeutil.UniqueID) string {
+	return "collection_" + strconv.FormatInt(collectionID, 10)
+}
+
+// TODO: move to metrics.go
+func NodeID2Label(nodeID typeutil.UniqueID) string {
+	return strconv.FormatInt(nodeID, 10)
+}
+
+// dataNodeDurationBuckets involves durations in milliseconds
+var dataNodeDurationBuckets = []float64{0, 10, 100, 1000, 10000, 100000, 1000000, 10000000}
+
+var (
+	DataNodeNumFlowGraphs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_flow_graphs",
+			Help:      "Number of flow graphs in DataNode.",
+		}, []string{
+			"num_flow_graphs",
+			nodeIDLabelName,
+		})
+
+	DataNodeConsumeMsgRowsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "message_size_count",
+			Help:      "Messages size count consumed from msgStream in DataNode.",
+		}, []string{
+			"message_size_count",
+			nodeIDLabelName,
+		})
+
+	DataNodeFlushedSize = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "flushed_size",
+			Help:      "Data size flushed to storage in DataNode.",
+		}, []string{
+			"flushed_size",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumDmlChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_dml_channels",
+			Help:      "Number of dmlChannels per collection in DataNode.",
+		}, []string{
+			"num_dml_channels",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumDeltaChannels = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_delta_channels",
+			Help:      "Number of deltaChannels per collection in DataNode.",
+		}, []string{
+			"num_delta_channels",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumConsumers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_consumers",
+			Help:      "Number of consumers per collection in DataNode.",
+		}, []string{
+			"num_consumers",
+			nodeIDLabelName,
+		})
+
+	DataNodeNumProducers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: milvusNamespace,
+			Subsystem: typeutil.DataNodeRole,
+			Name:      "num_producers",
+			Help:      "Number of producers per collection in DataNode.",
```

Comment:
add collection_id label to distinguish different collections.

Reply:
fixed